lib/coap/transport.ex

```diff
@@ -0,0 +1,192 @@
+defmodule CoAP.Transport do
+  @moduledoc """
+  Implements CoAP message layer
+  """
+  use GenStateMachine
+
+  alias CoAP.Message
+
+  @ack_timeout 2_000
+  @ack_random_factor 1.5
+  @max_retransmit 4
+
+  defstruct socket: nil,
```

Comment:
What are your thoughts about having a GenServer also be a Struct. I have sometimes made a module just called `State` to hold the struct. Not sure that it matters.

Reply:
My habit is to use dedicated module only when there is a need to add some helper functions (like `Connection.State.add_options/2`
Apart from that, I have no opinion, only habits :)

Reply:
I'm wondering if the options parsing in init can go into the `State`.

Reply:
I've moved all state handling related code to separate module